### ISSUE TITLE
Make all the nodes available through `get_node`

### ISSUE THREAD

**Issue by pepoviola** [paritytech/zombienet-sdk#127]
The other thing is: `get_node` only return validator node for now, should we adapt the code like this so that we can get reference to any node, including parachain collators?

`add_para` probably need to be adjusted like this 
```diff
     pub(crate) fn add_para(&mut self, para: Parachain) {
+        para.collators.iter().for_each(|collator| {
+            self.nodes_by_name
+                .insert(collator.name.clone(), collator.clone());
+        });
         self.parachains.insert(para.para_id, para);
     }
```

_Originally posted by @pgherveou in https://github.com/paritytech/zombienet-sdk/issues/126#issuecomment-1794845510_

**Comment by pgherveou** [comment]
This is my branch: https://github.com/paritytech/zombienet-sdk/pull/129

**Comment by pepoviola** [comment]
Yes, this is a `bug` tracked here https://github.com/paritytech/zombienet-sdk/issues/120 . @wirednkod dod you think is easily fixed?
